src/allotropy/allotrope/schema_parser/backup_manager.py

```diff
@@ -17,19 +19,32 @@ def _files_equal(path1: PathType, path2: PathType) -> bool:
 
 def _get_backup_path(path: PathType) -> Path:
     _path = Path(path)
-    return Path(_path.parent, f".{_path.stem}.bak{_path.suffix}")
+    return Path(_path.parent, f".{_path.stem}.{random_uuid_str()}.bak{_path.suffix}")
+
+
+def get_original_path(path: PathType) -> Path:
+    _path = Path(path)
+    filename = _path.stem
+    if filename.startswith("."):
+        filename = f"{Path(Path(filename[1:]).stem).stem}{_path.suffix}"
```

Comment:
this is not a legible line of code 🤣 but looks ok

Reply:
Who me? Write illegible code?!? I never :P

Reply:
(added a comment)